server/src/main/java/keywhiz/service/resources/automation/v2/ExpirationExtractor.java

```diff
@@ -17,11 +21,57 @@
 import org.bouncycastle.util.io.pem.PemReader;
 
 import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Collections.list;
 
 /** Helper class to extract expirations from secrets contents (best effort) */
 public final class ExpirationExtractor {
   private ExpirationExtractor() {}
 
+  public static Instant expirationFromKeystore(String type, String password, byte[] content) {
+    KeyStore ks;
+    try {
+      ks = KeyStore.getInstance(type);
+    } catch (KeyStoreException e) {
+      // Should never occur (assuming JCE is installed)
+      throw Throwables.propagate(e);
+    }
+
+    try {
+      ks.load(new ByteArrayInputStream(content), password.toCharArray());
+    } catch (IOException | NoSuchAlgorithmException | CertificateException e) {
+      // Failed to parse
```

Comment:
It returns true/false to the caller. I can log the exception if you want, but I'd rather not warn.